chain/network/src/routing.rs

```diff
@@ -355,9 +355,9 @@ impl RoutingTable {
     }
 
     pub fn contains_account(&self, announce_account: AnnounceAccount) -> bool {
-        self.account_peers
-            .get(&announce_account.account_id)
-            .map_or(false, |cur_announce_account| *cur_announce_account == announce_account)
+        self.account_peers.get(&announce_account.account_id).map_or(false, |cur_announce_account| {
+            cur_announce_account.peer_id == announce_account.peer_id
```

Comment:
Why not check the account id is also the same?

Reply:
Because `account_id` must coincide given that we get it from a map that maps `account_id` to `account_announcement`. If we check that, it is more a sanity check!

Reply:
In that case we should probably panic or return some sort of error when account id does not match.